Mathlib/AlgebraicGeometry/AffineScheme.lean

```diff
@@ -365,11 +364,12 @@ theorem SpecΓIdentity_hom_app_fromSpec :
       (Spec Γ(X, U)).presheaf.map (eqToHom hU.fromSpec_preimage_self).op := by
   have : IsAffine _ := hU
   delta IsAffineOpen.fromSpec Scheme.isoSpec
-  rw [Scheme.comp_app, Scheme.comp_app, ΓSpecIso_obj_hom,
-    Scheme.ofRestrict_app_self]
-  simp only [Category.assoc]
-  dsimp only [asIso_inv, Functor.op_obj, unop_op]
-  rw [← Functor.map_comp_assoc, ← op_comp, eqToHom_trans, Scheme.eq_restrict_presheaf_map_eqToHom,
+  rw [Scheme.comp_app, Scheme.comp_app, ΓSpecIso_obj_hom, Scheme.Opens.ι_app_self]
+  simp only [asIso_inv, Scheme.comp_coeBase, Opens.map_comp_obj, Scheme.Opens.topIso_inv,
+    Opens.map_top, Functor.id_obj, Functor.comp_obj,
+    Functor.rightOp_obj, Scheme.Γ_obj, unop_op, Scheme.Spec_obj, Scheme.Opens.topIso_hom,
+    Category.assoc]
+  rw [← Functor.map_comp_assoc, ← op_comp, eqToHom_trans, Scheme.Opens.eq_presheaf_map_eqToHom,
```

Comment:
The slowdown on this file comes from this proof and it will be improved in a later PR where the unit of the gamma-spec adjunction is specialized and be provided dedicated API.

Reply:
This whole proof can be solved by a single `simp only [many things are here]` and the `have` can be deleted.

Reply:
Do you mean introducing intermediate lemmas? I'm not sure how to make this work.

Reply:
Okay I managed to condense it into two blobs of simp and removed the nolint and the maxheartbeat.

Reply:
By the way does `rw` unfolds things that `simp` does not? I though they look through the same transparency level (both can see through reducible defs but not semireducible ones) but am I wrong?

Reply:
I *think* `rw` can see through `semi-reducible` terms. The discrimination tree for `simp` is pretty good at filtering things that are only defeq at default transparency.

Reply:
Hmm. I think it is *supposed* to operate at `reducible` but I can't tell if it is doing anything atm

Reply:
The actual substitution via `abstract` is done at `reducible` transparency but there is at least one other `isDefEq` call that is done at `default`. So we start unfolding these more than is desired here

Reply:
Is that intended? If this happens to `rw` everywhere, it sounds like a performance hazard to me.

Reply:
With the huge caveat that I just looked at the file for a couple of minutes, no.

Reply:
I'll check on this at office hours tomorrow at the latest